lib/proximity.js

```diff
@@ -448,41 +459,80 @@ function Proximity(opts) {
      */
     inches: {
       get: function() {
-        return +(this.centimeters * 0.39).toFixed(2);
+        return +(this.centimeters * 0.39).toFixed(3);
       }
     },
     in: {
       get: function() {
         return this.inches;
       }
     },
+    filtered: {
+      get: typeof opts.filter === "function" ? opts.filter : function() {
+        var average = this.samples.reduce(function(a, b) {
+          return a + b;
+        }) / this.samples.length;
+        var isSoften = typeof filterOpts.soften === "number";
+
+        raw = average;
+
+        return {
+          cm: isSoften ? soften(this.cm) : this.cm,
+          centimeters: isSoften ? soften(this.centimeters) : this.centimeters,
+          in : isSoften ? soften(this.in) : this.in,
+          inches: isSoften ? soften(this.inches) : this.inches,
+        };
+
+        function soften(num) {
+          return Math.round(num * Math.pow(10, filterOpts.soften)) / Math.pow(10, filterOpts.soften);
+        }
+      }
+    }
   });
 
   if (typeof this.initialize === "function") {
     opts.pinValue = pinValue;
     this.initialize(opts, function(data) {
+
+      if (isFiltered && typeof filterOpts.deviation === "number") {
+        data = raw && data > raw * (filterOpts.deviation) ? raw : data;
```

Comment:
Ok... I think that's going to be problematic. We're trying to hide the raw values because they are usually nonsensical as far as the user is concerned. Doing this here is also super invasive and makes it basically impossible to abstract this feature into something we can apply to all input things, which is definitely the goal. Maybe we just need more time to think about it...